src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfilesController.php

```diff
@@ -0,0 +1,161 @@
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Configure\AdvancedParameters;
+
+use PrestaShop\PrestaShop\Core\Search\Filters\ProfilesFilters;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use PrestaShopBundle\Security\Annotation\DemoRestricted;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class ProfilesController is responsible for displaying the
+ * "Configure > Advanced parameters > Team > Profiles" page.
+ */
+class ProfilesController extends FrameworkBundleAdminController
+{
+    /**
+     * Show profiles listing page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @Template("@PrestaShop/Admin/Configure/AdvancedParameters/Profiles/profiles.html.twig")
+     *
+     * @param ProfilesFilters $filters
+     *
+     * @return array
+     */
+    public function indexAction(ProfilesFilters $filters)
+    {
+        $profilesGridFactory = $this->get('prestashop.core.grid.factory.profiles');
+        $gridPresenter = $this->get('prestashop.core.grid.presenter.grid_presenter');
+
+        return [
+            'layoutHeaderToolbarBtn' => [
+                'add' => [
+                    'href' => $this->getAdminLink('AdminProfiles', ['addprofile' => '']),
+                    'desc' => $this->trans('Add new profile', 'Admin.Advparameters.Feature'),
+                    'icon' => 'add_circle_outline',
+                ],
+            ],
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink('AdminProfiles'),
+            'grid' => $gridPresenter->present($profilesGridFactory->getGrid($filters)),
+        ];
+    }
+
+    /**
+     * Used for applying filtering actions.
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $definitionFactory = $this->get('prestashop.core.grid.definition.factory.profiles');
+        $definitionFactory = $definitionFactory->getDefinition();
+
+        $gridFilterFormFactory = $this->get('prestashop.core.grid.filter.form_factory');
+        $searchParametersForm = $gridFilterFormFactory->create($definitionFactory);
+        $searchParametersForm->handleRequest($request);
+
+        $filters = [];
+
+        if ($searchParametersForm->isSubmitted()) {
+            $filters = $searchParametersForm->getData();
+        }
+
+        return $this->redirectToRoute('admin_profiles', ['filters' => $filters]);
+    }
+
+    /**
+     * Shows profile edit form.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
```

Comment:
Provide redirectRoute parameter

```suggestion
     *     "is_granted('update', request.get('_legacy_controller'))",
     *     redirectRoute="admin_profiles",
```